fleet_vehicle_inspection/models/fleet_vehicle_inspection.py

```diff
@@ -0,0 +1,196 @@
+        "inspection_id",
+        string="Inspection Lines",
+        copy=True,
+        auto_join=True,
+        states=READONLY_STATES,
+    )
+
+    result = fields.Selection(
+        [("todo", "Todo"), ("success", "Success"), ("failure", "Failure")],
+        "Inspection Result",
+        default="todo",
+        compute="_compute_inspection_result",
+        readonly=True,
+        copy=False,
+        store=True,
+    )
+
+    @api.depends("inspection_line_ids", "state")
+    def _compute_inspection_result(self):
+        for inspection in self:
+            if inspection.inspection_line_ids:
+                if inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "todo"
+                ):
+                    inspection.result = "todo"
+                    continue
+                elif inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "failure"
+                ):
+                    inspection.result = "failure"
+                else:
+                    inspection.result = "success"
+            else:
+                inspection.result = "todo"
+
+    @api.model
+    def create(self, vals):
+        if vals.get("name", "New") == "New":
+            if vals.get("direction") == "out":
+                vals["name"] = (
+                    self.env["ir.sequence"].next_by_code("fleet.vehicle.inspection.out")
+                    or "/"
+                )
+            else:
+                vals["name"] = (
+                    self.env["ir.sequence"].next_by_code("fleet.vehicle.inspection.in")
+                    or "/"
+                )
+        return super(FleetVehicleInspection, self).create(vals)
+
+    @api.multi
+    def button_cancel(self):
+        for inspection in self:
+            if inspection.state not in ["draft", "confirmed"]:
+                continue
+            inspection.write({"state": "cancel"})
+        return True
+
+    @api.multi
+    def button_confirm(self):
+        for inspection in self:
+            if inspection.inspection_line_ids:
+                if inspection.state not in ["draft", "cancel"]:
+                    continue
+                if inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "todo"
+                ):
+                    raise UserError(
+                        _(
+                            "Inspection cannot be completed. "
+                            "There are uninspected items."
+                        )
+                    )
+                inspection.write({"state": "confirmed"})
+            else:
+                raise UserError(
+                    _("Inspection cannot be completed. There are no inspected items.")
+                )
+        return True
+
+    @api.multi
+    def button_draft(self):
+        for inspection in self:
+            inspection.write({"state": "draft"})
+            inspection.write({"result": "todo"})
+        return True
```

Comment:
You can minimize the number of writes :)

```suggestion
        return self.write({
            "state": "draft",
            "result": "todo",
        })
```